### ISSUE TITLE
Allow to select either Onlyoffice or Collabora if both suites are installed on the Nextcloud server

### ISSUE THREAD

**Issue by e-alfred** [nextcloud/android#5729]
### Steps to reproduce
1. Open ODT or DOCX file or other supported document types.

### Expected behaviour
- In the context menu, an option to select "Open with Collabora" and "Open with Onlyoffice" should show up if both office suites are installed on the Nextcloud server. This is more consistent with the web interface of Nextcloud.

### Actual behaviour
- Currently ODT is opened with Collabora and DOCX gets opened with Onlyoffice. Onlyoffice only allows viewing of the document because of the removal of editing features in their Open Source version.

### Environment data
Android version: Lineageos 16.1/Android 9

Device model: BQ Aquaris X Pro

Stock or customized system: Custom ROM

Nextcloud app version: 3.10

Nextcloud server version: 18.0.3

**Comment by jancborchardt** [comment]
We had this discussion before when we added it to the desktop app, see https://github.com/nextcloud/desktop/issues/1564

Excerpt from that:

> - We shouldn’t expose the app name as this is not really relevant to regular people.
> - It’s focused on Collabora Online, OnlyOffice and Text, and if there are multiple apps to edit with it should just pick 1 for simplicity.

We don’t want this to grow into a potentially endless list of apps to edit with. Even 2 is too much – it should just work, especially on mobile.
People wouldn’t know on what basis to decide anyway, so we should pick the proper option.

cc @juliushaertl @karlitschek @misch7 for reference.

**Comment by e-alfred** [comment]
Simplicity is fine, but Onlyoffice for example doesn't allow to edit on mobile anymore in their community version (very "nice" move from them) and Collabora doesn't always display things correctly. Sometimes one solution doesn't provide the features needed, so having a choice is better than limiting them. To make all options available, the user has the maximum options to do their work and to choose their favorite tool for the task at hand. If it is just too much for users to handle, the admin can set limitations if needed.

Even simple to use mobile operating systems (iOS and Android) allow you to select all registered apps for a certain task to select from and this seems to work well enough.

**Comment by immanuelfodor** [comment]
The integration edition or the contributor version of OO still offers edit mode on mobile, it'd be great to keep it working for those who have such licenses. I'd be happy with an app chooser if multiple apps are available.

**Comment by e-alfred** [comment]
Yes, an "Open with" option (present on all major OS inclunding Linux, Windows, macOS, iOS, Android) is probably the best variant and pretty much the norm out there.

**Comment by e-alfred** [comment]
Yes, an "Open with" option (present on all major OS inclunding Linux, Windows, macOS, iOS, Android) is probably the best variant and pretty much the norm out there.

Given that every editing app (Onlyoffice, Collabora CODE, Text) have their advantages and disadvantages and none are a "it just works" option at all limiting the choice of the "right tool for the right job" is the opposite of a productive environment because users are forced to use an inadequate tool with no option to change that.

**Comment by juliushaertl** [comment]
> We don’t want this to grow into a potentially endless list of apps to edit with. Even 2 is too much – it should just work, especially on mobile.
> People wouldn’t know on what basis to decide anyway, so we should pick the proper option.

I'm not sure we can here, since both OnlyOffice and Collabora might have their benefits depending on the document format, the changes you plan to do and how you want to work on a document. The default will be that people only have one Office solution integrated, so in those cases we can just show "Edit" there, if two or more are installed, we can pick a default opener, but we should still show both options in the context/3-dots menu. If someone decides to have two different editors enabled, we should not artificially block them from using one or the other imo.

**Comment by e-alfred** [comment]
@juliushaertl Thanks for your input, just what I was thinking.

**Comment by jancborchardt** [comment]
> Yes, an "Open with" option (present on all major OS inclunding Linux, Windows, macOS, iOS, Android) is probably the best variant and pretty much the norm out there.

We are not an operating system but a productivity suite. It’s better to compare it e.g. to Google Drive / Docs where you also don’t have to choose between what you open a document with – you get the best tool for the job.

It makes 0 sense to open Markdown files with anything else than Text, and you also can’t edit odt and such with Text. So Text is already separate and will not be part of any "Open with" dialog.

As for Collabora Online and OnlyOffice – they do the exact same thing, so I’m curious why both would be installed in the first place?

-------------

If we do anything of the sort:
- Keep in mind this needs to be maintained
- It is only for a very special case
- The default click should still work to open the document with one click, not blocked by a choice of tool.
- If at all, any other tools should be listed in the action menu as "Open with …". This should not grow into an endless list, nor should e.g. Markdown files have OnlyOffice or so listed if Text is installed.

cc @karlitschek @jospoortvliet your input please.

**Comment by e-alfred** [comment]
This is a difficult issue I guess.

> We are not an operating system but a productivity suite.

Absolutely, but not every tool is a perfect fit for a certain job, so maybe trying another one is surely not that wrong? Having a few options is sometimes more productive than being limited to a single solution. If it should be limited, then the admin could do that in the permission settings.

>  to Google Drive / Docs where you also don’t have to choose between what you open a document with

Well, Google wants you to use Google stuff of course (vendor lock in and of course more data for Doubleclick, I guess), but even Dropbox allows to select different office suites (Google Docs and Microsoft Office Online):

![image](https://user-images.githubusercontent.com/10073629/78075251-7865f500-73a4-11ea-9e59-1d0788ef150d.png)

I think such kind of a menu would solve the problem pretty nicely, I guess?

> It makes 0 sense to open Markdown files with anything else than Text, and you also can’t edit odt and such with Text. So Text is already separate and will not be part of any "Open with" dialog.

For Markdown files, this is probably true (but I would say not even in this case, many people prefer the files_markdown app for that, so even an issue in this case), but what about CSV files for example? 

They work with each type of editor (Text, Collabora, Onlyoffice), but each of them has advantages and disadvantages. Maybe I want to edit it raw? Then I go for Text. Meanwhile, if I open it in Collabora, I can't select the delimiter, so everything is in a single cell, while in Onlyoffice it works well (showing a popup that "blocks" opening the file to let me select the right delimiter). 

If I want to edit an office document on a mobile device, well, only Collabora can do that after Onlyoffice removed this feature.

> As for Collabora Online and OnlyOffice – they do the exact same thing, so I’m curious why both would be installed in the first place?

In theory, yes, but in practice sadly not. They all have their limits and advantages (I mentioned a few above, surely more complex documents cause a bunch more). One more option is better than nothing, I would say.

**Comment by karlitschek** [comment]
I'm unsure. There are benefits for both sides. I like the simplicity to only have one option. But there might be different apps that can handle the same document type. Of course this could theoretically be configured on the server. So I'm unsure. Sorry.

**Comment by immanuelfodor** [comment]
Even when you download a file in the NC Android app, you have an option "Open with" in the file's three dot menu, so I don't really get why this is such a big problem to have an "Edit with" option. After tapping a PDF, for example, the file gets downloaded, and Android also asks which app do I want to use (Drive PDF, Acrobat Reader, etc). The same dialog should be presented to the user when tapping the file and the "Edit with" option for code reuse in the NC app. If the NC server admin sets a default editor app, then tapping a file should open that editor's web view, and any "advanced" user can choose an editor to their liking from the "Edit with" menu. Seems not so complicated after all :)

**Comment by pirate** [comment]
Bumping this. We run both OnlyOffice and Collabora on our Nextcloud server because some of our clients prefer OnlyOffice for faster Desktop editing, and others prefer Collabora for a better mobile experience.

We'd like to set the organization-wide default to open with OnlyOffice on single click, but have an option to open with Collabora under the "three dots" dropdown context menu.

Something like this would be ideal:  <img src="https://github.com/nextcloud/android/assets/511499/4b417dfb-98ba-4614-9773-a877d9401425" width="400px" alt="UI mockup"/>
